sales/src/execute.rs

```diff
@@ -125,24 +139,36 @@ where
             // or the supply cap is 0 (unlimited)
             {
                 // check if enough fee was sent
-                if info.funds.len() == 0 {
+                if info.funds.is_empty() {
                     return Err(ContractError::NoFundsError);
                 } else if info.funds.len() > 1 {
                     return Err(ContractError::MultipleFundsError);
                 } else {
                     if info.funds[0].denom.ne(&sale.price[0].denom) {
                         return Err(ContractError::WrongFundError);
                     }
-                    if info.funds[0].amount.lt(&info.funds[0].amount) {
+                    if info.funds[0].amount.lt(&sale.price[0].amount) {
```

Comment:
We made an error comparing the sent funds with itself instead of with the sale price